src/extension.ts

```diff
@@ -64,7 +65,7 @@ export async function activate(context: vscode.ExtensionContext, perfStats: { lo
         activateContext.telemetry.measurements.mainFileLoad = (perfStats.loadEndTime - perfStats.loadStartTime) / 1000;
 
         setupEvents(context);
-        ext.subscriptionProvider = new VSCodeAzureSubscriptionProvider(context.globalState);
+        ext.subscriptionProviderFactory = ext.isWeb ? createWebSubscriptionProviderFactory(context) : createAzureAccountSubscriptionProviderFactory();
```

Comment:
```suggestion
        ext.subscriptionProviderFactory ??= ext.isWeb ? createWebSubscriptionProviderFactory(context) : createAzureAccountSubscriptionProviderFactory();
```

Might be useful for unit tests? I dunno.

Reply:
For unit tests, what's worked best for me is adding a `ext.overrideBlahFactory`.